svelte/src/scripts/draw/helper/custom-d3-forces.ts

```diff
@@ -1,166 +1,222 @@
+    };
+    
+    function calcForceStrength(node1: squareData, node2: squareData, alpha: number) {
+        return (node1.width + node1.height) / (node1.width + node1.height + node2.width + node2.height)
+            * (1 + alpha)
+            * 20; // random magic number
+    }
+
+    // actual force calulation
 	function force(alpha: number) {
-		// Loop through all nodes for collision check
+        // Loop through all nodes for collision check
 		for (let i = 0; i < nodes.length; i++) {
 			for (let j = i + 1; j < nodes.length; j++) {
-				if (nodes[i].x && nodes[i].y && nodes[j].x && nodes[j].y) {
-					// Caclulate x and y position of the top-left and bottom-right corner of the node, and middle position.
-					const node1 = makeSquare(nodes[i]);
-					const node2 = makeSquare(nodes[j]);
-
-					// Check for collision
-					if (
-						node1.bottomright.x >= node2.topleft.x &&
-						node1.topleft.x <= node2.bottomright.x &&
-						node1.bottomright.y >= node2.topleft.y &&
-						node1.topleft.y <= node2.bottomright.y
-					) {
-						// We collided!
-						// We want small nodes to move out of the way for large nodes
-						// Also, the strenght of the force should decrease with alpha
-						const n1ForceStrength = calcForceStrength(node1, node2, alpha);
-						const n2ForceStrength = calcForceStrength(node2, node1, alpha);
-
-						// Trig stuff to calculate the direction of the force-vector
-						const delta = (node1.middle.y - node2.middle.y) / (node1.middle.x - node2.middle.x);
-						const angle = Math.atan(delta);
-						const vy = Math.sin(angle);
-						const vx = Math.cos(angle);
-
-						// Set the values
-						if (node1.middle.x > node2.middle.x) {
-							nodes[i].vx! += vx * n1ForceStrength;
-							nodes[i].vy! += vy * n1ForceStrength;
-							nodes[j].vx! -= vx * n2ForceStrength;
-							nodes[j].vy! -= vy * n2ForceStrength;
-						} else {
-							nodes[i].vx! -= vx * n1ForceStrength;
-							nodes[i].vy! -= vy * n1ForceStrength;
-							nodes[j].vx! += vx * n2ForceStrength;
-							nodes[j].vy! += vy * n2ForceStrength;
-						}
-					}
-				}
+                if (nodes[i].x && nodes[i].y && nodes[j].x && nodes[j].y) {
+                    // Caclulate x and y position of the top-left and bottom-right corner of the node, and middle position.
+                    const node1 = makeSquare(nodes[i]);
+				    const node2 = makeSquare(nodes[j]);
+                    
+                    // Check for collision
+                    if (node1.bottomright.x >= node2.topleft.x && node1.topleft.x <= node2.bottomright.x && 
+                        node1.bottomright.y >= node2.topleft.y && node1.topleft.y <= node2.bottomright.y) {
+                        // We collided!
+                        if (snap) {
+                            // Snap stuff into place
+                            if (node2.middle.x < node1.middle.x) {
+                                nodes[i].x = nodes[j].x! + 0.5 * nodes[j].width + 0.5 * nodes[i].width;
```

Comment:
my brain is not big enough to visualize what is the meaning of this :rofl: . I'll let it go and just try to understand it based on the behavior in the browser later

Reply:
This is dead code anyway (as mentionen in the OP), but it should just move the node out of the other node. Goes wrong in combination with cx and cy though.